code/game/objects/items/devices/spyglasses.dm

```diff
@@ -1,64 +1,305 @@
+
+	var/disabled = FALSE
+
+/obj/item/video_bug/Initialize(mapload)
+	update_overlays()
+	var/tile_overlay = image(icon = icon, icon_state = icon_state_clothing, pixel_x = rand(-12, 12), pixel_y = rand(-12, 12))
+	AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE, INVISIBILITY_OBSERVER, tile_overlay, use_anchor = TRUE)
+	bug_id = "[pick(GLOB.posibrain_names)]-[rand(1,999)]"
+	GLOB.video_bug_list.Add(src)
+	create_view()
+	. = ..()
+
+/obj/item/video_bug/Destroy()
+	QDEL_NULL(cam_screen)
+	QDEL_LIST(cam_plane_masters)
+	QDEL_NULL(tracker)
+	QDEL_NULL(cam_background)
+	GLOB.video_bug_list.Remove(src)
+	if(attached_to)
+		attached_to.tracking_bug = null
+		attached_to.update_overlays()
+		attached_to = null
+	if(viewing_glasses)
+		viewing_glasses.viewed_bug = null
+		viewing_glasses = null
+	return ..()
+
+/obj/item/video_bug/examine(mob/user)
+	. = ..()
+	if(bug_network_id)
+		. += "<span class='notice'>It is connected to \the <b>[bug_network_id]</b> network.</span>"
+	else
+		. += "<span class='warning'>It is not connected to any network!</span>"
+	if(bug_id)
+		. += "<span class='notice'>It's ID is <b>[bug_id]</b></span>"
+	else
+		. += "<span class='warning'>It does not have an ID!</span>"
+
 
-/obj/item/clothing/accessory/spy_bug/Initialize(mapload)
+/obj/item/video_bug/update_overlays()
 	. = ..()
+	if(!disabled)
+		. += mutable_appearance(icon, "camerabug_overlay")
+		. += emissive_appearance(icon, "camerabug_overlay_emissive", alpha = alpha)
+
+/obj/item/video_bug/proc/attach_to_clothing_item(mob/user, var/obj/item/clothing/target, instant = FALSE)
+	if(!target.buggable || target.item_flags & ABSTRACT)
+		to_chat(user, "<span class='warning'>You can't attach \the [src] to \the [target]!</span>")
+		return FALSE
+	if(target.tracking_bug)
+		to_chat(user, "<span class='warning'>\The [target] already has something attached!</span>")
+		return FALSE
+	if(!instant)
+		to_chat(user, "<span class='notice'>You start to carefully plant [src] to \the [target].</span>")
+		if(!do_after(user, 1 SECONDS, get_turf(target)))
+			to_chat(user, "<span class='warning'>Your hand slips and you decide not to plant \the [src]!</span>")
+			return FALSE
+	var/left_border = 1
+	var/right_border = 32
+	var/bottom_border = 1
+	var/top_border = 32
+	var/list/temp_list = target.get_bounding_box()
+	left_border = temp_list["left"]
+	right_border = temp_list["right"]
+	bottom_border = temp_list["bottom"]
+	top_border = temp_list["top"]
+	var/icon/target_icon = getFlatIcon(target)
+	var/attempts = 0
+	var/selected_x
+	var/selected_y
+	var/selected_coordinates = FALSE
+	while(attempts < 10)
+		selected_x = rand(left_border + 2, right_border - 2)
+		selected_y = rand(bottom_border + 2, top_border - 2)
+		if(isnull(target_icon.GetPixel(selected_x, selected_y)))
+			attempts ++
+		else
+			selected_coordinates = TRUE
+			break
+	if(!selected_coordinates)
+		selected_x = 15
+		selected_y = 15
+	user.temporarilyRemoveItemFromInventory(src)
+	forceMove(null)
```

Comment:
This feels like it is going to cause loads of issues, like with the cryo pod. You can probably make stuff like the cryo-pod handled better by making this a component instead and just qdeling the bug and referencing the component instead.
A majority of this code is written in a way that wants to be a component but is suffering from not being a component and is forced to insert specific type checks all over the codebase to componensate. This code needs to be modular and a component would help with that.